html/modules/custom/reliefweb_import/src/Plugin/ReliefWebImporter/EchoDataImporter.php

```diff
@@ -0,0 +1,391 @@
+    ksort($documents);
+
+    // Process the documents importing new ones and updated ones.
+    $processed = $this->processDocuments($documents, $provider_uuid, $plugin);
+
+    // @todo check if we want to return TRUE only if there was no errors or if
+    // return TRUE for partial success is fine enough.
+    return $processed > 0;
+  }
+
+  /**
+   * Retrieve documents from the Echo Flash API.
+   *
+   * @param int $limit
+   *   Maximum number of documents to retrieve at once.
+   * @param string $order_property
+   *   Property to use to sort the documents.
+   *
+   * @return array
+   *   List of documents keyed by IDs.
+   */
+  protected function getDocuments(int $limit, string $order_property = 'created'): array {
+    // Get list of documents.
+    try {
+      $timeout = $this->getPluginSetting('timeout', 10, FALSE);
+      $api_url = $this->getPluginSetting('api_url');
+
+      $response = $this->httpClient->get($api_url, [
+        'connect_timeout' => $timeout,
+        'timeout' => $timeout,
+      ]);
+
+      if ($response->getStatusCode() !== 200) {
+        // @todo try to retrieve the error message.
+        throw new \Exception('Failure with response code: ' . $response->getStatusCode());
+      }
+
+      $content = $response->getBody()->getContents();
+
+      if (!empty($content)) {
+        $documents = json_decode($content, TRUE, flags: \JSON_THROW_ON_ERROR);
+      }
+      else {
+        return [];
+      }
+    }
+    catch (\Exception $exception) {
+      $message = $exception->getMessage();
+      throw new \Exception($message);
+    }
+
+    if (!isset($documents['Items'])) {
+      return [];
+    }
+
+    // Map the document's data to the document's ID.
+    $map = [];
+    foreach ($documents['Items'] as $document) {
+      if (!isset($document['ContentItemId'])) {
+        continue;
+      }
+      $map[$document['ContentItemId']] = $document;
+    }
+
+    return $map;
+  }
+
+  /**
+   * Process the documents retrieved from the Echo Flash API.
+   *
+   * @param array $documents
+   *   Echo Flash documents.
+   * @param string $provider_uuid
+   *   The provider UUID.
+   * @param \Drupal\reliefweb_post_api\Plugin\ContentProcessorPluginInterface $plugin
+   *   The Post API content plugin processor used to import the documents.
+   *
+   * @return int
+   *   The number of documents that were skipped or imported successfully.
+   */
+  protected function processDocuments(array $documents, string $provider_uuid, ContentProcessorPluginInterface $plugin): int {
+    $schema = $this->getJsonSchema('report');
+
+    // This is the list of extensions supported by the report attachment field.
+    $extensions = explode(' ', 'csv doc docx jpg jpeg odp ods odt pdf png pps ppt pptx svg xls xlsx zip');
+    $allowed_mimetypes = array_filter(array_map(fn($extension) => $this->mimeTypeGuesser->guessMimeType('dummy.' . $extension), $extensions));
+    $allowed_mimetypes[] = 'application/octet-stream';
```

Comment:
I don't think flash updates have attachments (see https://reliefweb.int/updates?search=source%3AECHO+AND+origin%3A*Echo-Flash*+AND+NOT+_exists_%3Afile with and with out the `NOT _exists_). There is just one with an attachment and it was probably not a flash update in the first place.

So I think everything related to attachments could be removed from this importer to simplify it.